pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,223 @@ func TestAccessTokenFile(t *testing.T) {
 
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
-
-			// Test we are reading the new Data
-			dataExpected, err := tokenInFile.Read()
-			eq := reflect.DeepEqual(dataActual, dataExpected)
-
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+			Convey("When running the Write method", func() {
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Writes the access token without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Creates with file without raising an error", func() {
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite2")
+					So(err, ShouldEqual, nil)
+				})
+			})
+
+			Convey("When running the Write method a second time", func() {
+				dataActual = []byte{'t', 'e', 's', 't', '2'}
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Creates with file without raising an error", func() {
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite2")
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Writes the data the file as expected", func() {
+					dataExpected, _ := tokenInFile.Read()
+					eq := reflect.DeepEqual(dataActual, dataExpected)
+					So(eq, ShouldEqual, true)
+				})
+			})
 		})
 	})
 
 	Convey("Delete", t, func() {
-		Convey("Returns no error after read write and delete", func() {
-			dataActual := []byte{'t', 'e', 's', 't'}
-			tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel1"
-			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
-
-			dataFromRead, err := tokenInFile.Read()
-			So(err, ShouldEqual, nil)
-
-			err = tokenInFile.Delete()
-			So(err, ShouldEqual, nil)
 
-			// Check that file does not exits
-			_, err = os.Stat("/tmp/accessTokenFileDel1")
-			So(err, ShouldNotEqual, nil)
+		Convey("Given access token with data", func() {
+			dataActual := []byte{'t', 'e', 's', 't'}
 
-			// Both input & output arrays should be cleared from memory
-			empty := make([]byte, len(dataActual))
-			eq := reflect.DeepEqual(dataActual, empty)
-			So(eq, ShouldEqual, true)
-			eq = reflect.DeepEqual(dataFromRead, empty)
-			So(eq, ShouldEqual, true)
+			Convey("When running the Write method", func() {
+				tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel1"
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("When running the Read method", func() {
+					dataFromRead, err := tokenInFile.Read()
+
+					Convey("Finishes without raising an error", func() {
+						So(err, ShouldEqual, nil)
+					})
+
+					Convey("When running the Delete method", func() {
+						err = tokenInFile.Delete()
+
+						Convey("Finishes without raising an error", func() {
+							So(err, ShouldEqual, nil)
+						})
+
+						Convey("Properly deletes the file", func() {
+							_, err = os.Stat("/tmp/accessTokenFileDel1")
+							So(err, ShouldNotEqual, nil)
```

Comment:
do we want to check the actual error? to verify it doesn't fail oin something else and the file is not deleted

Reply:
adding but what other errors could it be besides `error deleting access token` (that's what we wrote in `Delete`)? Maybe os specific?

Reply:
yes, os specific